### PR TITLE
Add Invalidate endoint

### PR DESCRIPTION
**Pull request by AleDore**
This PR enables the `/invalidate` endpoint in order to allow token blacklist over logout operations.
At the same time `introspect` endpoint adds some logic to recognize if a token is already blacklisted, before returning info about its state.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -121,17 +117,26 @@ const acs: AssertionConsumerServiceT = async user => {
...
-            .chain(token =>
-              setWithExpirationTask(
-                redisClient,
-                `${SESSION_TOKEN_PREFIX}${token}`,
-                JSON.stringify(tokenUser),
-                3600
-              ).map(() => token)
-            )
+            .map(_ => ({
+              expiration: 3600,
```
Comment (BurnedMarshal) [inline src/app.ts:128]: Set this value as global `const` or as config parameter.

File: src/app.ts [hunk 2]
```diff
@@ -195,21 +200,71 @@ export const createAppTask = withSpid({
     });
   });
   withSpidApp.post("/introspect", async (req, res) => {
```
Comment (BurnedMarshal) [inline src/app.ts:202]: Does this API work only on _ non-jwt_ token?

File: src/app.ts [hunk 3]
```diff
@@ -195,21 +200,71 @@ export const createAppTask = withSpid({
...
+          () => res.status(200).json({ active: true })
         )
       )
       .map(tokenUser => ({ active: true, user: tokenUser }))
-      .fold(identity, _ => res.json(_))
+      .fold(identity, _ => res.status(200).json(_))
+      .run();
+  });
+
+  withSpidApp.post("/invalidate", async (req, res) => {
```
Comment (BurnedMarshal) [inline src/app.ts:260]: This API should works on 2 different way if the token is JWT or opaque.
JWT needs to store on redis the invalidate token
opaque token we can delete the existing token

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: src/app.ts (hunk 2 above)
Comment (AleDore) [reply src/app.ts:202]: Nope it is used also to check if a token (JWT or not) has been previously blacklisted

File: src/app.ts (hunk 3 above)
Comment (AleDore) [reply src/app.ts:260]: 👍

**Pull request merged**